c/tskit/trees.h

```diff
@@ -227,41 +227,418 @@ typedef struct {
+    this tree sequence.
+
+TSK_BUILD_INDEXES
+    If specified edge indexes will be built and stored in the table collection
+@endrst
+
+@param self A pointer to an uninitialised tsk_table_collection_t object.
+@param options Allocation time options. Currently unused; should be
+    set to zero to ensure compatibility with later versions of tskit.
+@return Return 0 on success or a negative value on failure.
+*/
 int tsk_treeseq_init(
     tsk_treeseq_t *self, tsk_table_collection_t *tables, tsk_flags_t options);
 
+/**
+@brief Load a tree sequence from a file path.
+
+@rst
+Loads the data from the specified file into this tree sequence.
+The tree sequence is also initialised.
+The resources allocated must be freed using
+:c:func:`tsk_treeseq_free` even in error conditions.
+
+If the file contains multiple tree sequences, this function will load
+the first. Please see the :c:func:`tsk_treeseq_loadf` for details
+on how to sequentially load tree sequences from a stream.
+
+If the TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set, the table collection is
+read without loading the reference sequence.
+
+**Options**
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_LOAD_SKIP_REFERENCE_SEQUENCE
+    Do not load reference sequence.
+
+**Examples**
+
+.. code-block:: c
+
+    int ret;
+    tsk_treeseq_t ts;
+    ret = tsk_treeseq_load(&ts, "data.trees", 0);
+    if (ret != 0) {
+        fprintf(stderr, "Load error:%s\n", tsk_strerror(ret));
+        exit(EXIT_FAILURE);
+    }
+
+@endrst
+
+@param self A pointer to an uninitialised tsk_treeseq_t object
+@param filename A NULL terminated string containing the filename.
+@param options Bitwise options. See above for details.
+@return Return 0 on success or a negative value on failure.
+*/
 int tsk_treeseq_load(tsk_treeseq_t *self, const char *filename, tsk_flags_t options);
+
+/**
+@brief Load a tree sequence from a stream.
+
+@rst
+Loads a tree seqeuence from the specified file stream. The tree sequence
+is also initialised. The resources allocated must be freed using
+:c:func:`tsk_treeseq_free` even in error conditions.
+
+If the stream contains multiple tree sequences, this function
+will load the next tree sequence from the stream. If the stream contains no
+more tree sequence definitions the error value :c:macro:`TSK_ERR_EOF` will
+be returned. Note that EOF is only returned in the case where zero bytes are
+read from the stream --- malformed files or other errors will result in
+different error conditions. Please see the
+:ref:`sec_c_api_examples_file_streaming` section for an example of how to
+sequentially load tree sequences from a stream.
+
+Please note that this streaming behaviour is not supported if the
+TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set.
+If the TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set, the tree sequence is
+read without loading the reference sequence.
+When attempting to read from a stream with multiple tree sequence definitions
```

Comment:
Either of what two options? I'm not following the last bit here.

We should note here somewhere that the stream can be an arbitrary fd (e.g. a socket), and we don't perform any seek operations.

Reply:
Fixed